### ISSUE TITLE
Capped collection tailable.stream() -> tailable.cursor() example?

### ISSUE THREAD

**Issue by bazineta** [Automattic/mongoose#4254]
Noticed this morning that stream() is now deprecated, and cursor() is the replacement. However, in attempting to port code that was using tailable().stream() against a capped collection, it's clear that I must be missing some fundamental concept, as the results are, well, bizarre. While it would seem from the API that it's a direct replacement for code that simply used .on('data'), .on('close'), .on('error'), etc., that does not appear to be the case.

Is there some example that I could reference regarding how a tailable() against a capped collection is supposed to work with the new cursor() method?

**Comment by bazineta** [comment]
As an example of our current and now deprecated implementation, we have the following (with apologies in advance to those mortally offended by CoffeeScript). The capped collection is one that various other actors place mail items into to be sent. This is the polling loop of the mail sender, who waits for work to arrive in the collection. Items have a 'done' boolean flag to indicate if they've been handled here.

```
poll = ->

  mail = Mail.find
    done: false
  .tailable true, config.poll.options
  .stream()

  log.info 'Mail stream opened'

  mail.on 'data', (item) ->

    send item, (err, sent) ->

      if err
        log.error {err}
      else
        log.info {item}, sent

      item.done = true
      item.save()

      return

    return

  mail.on 'error', (err) ->

    log.error {err} unless isTailEnd err
    mail.destroy()
    return

  mail.on 'close', ->

    log.warn 'Mail stream closed, waiting', config.poll.delay.humanize()
    _.delay poll, config.poll.delay.asMilliseconds()

    return

  return
```

The isTailEnd() method is the best solution we could find for #2922. It's not something we're proud of, frankly:

```
isTailEnd = (err) -> (err.name    is 'MongoError' and
                      err.message in ['cursor killed or timed out'
                                      'No more documents in tailed cursor'])
```

The tailable options we're using are:

```
 # Poll delay; if we encounter an error during streamed read of the mail
  # collection, back off for this duration and try again. Define options
  # for the tail cursor portion of the streamed read.

  poll:
    delay: moment.duration(1, 'minute')
    options:
      awaitdata:             true
      numberOfRetries:       15
      tailableRetryInterval: moment.duration(1, 'minute').asMilliseconds()
```

**Comment by vkarpov15** [comment]
I suspect this is just a dup of #4266, currently QueryCursor doesn't pass query options correctly, this will be fixed in 4.5.3.
